### PR TITLE
Aws VM cluster creation

### PR DESCRIPTION
**Pull request by yujunglo**
This feature is used to support AWS VM cluster creation and automatically register cluster into autoscaling group and also establish the interface to support container. 
@sbaogang @jinruh @nickdechant 

1. Create cluster
screen shot 2016-02-17 at 3 26 09 pm

2. Update the cluster
screen shot 2016-02-17 at 3 17 06 pm

3. Expand the advanced setting
screen shot 2016-02-17 at 3 15 45 pm

4. Add new user data config dialog
screen shot 2016-02-17 at 3 16 09 pm

5. Delete Cluster dialog
screen shot 2016-02-17 at 3 18 02 pm

Cloud resource management view:
1. Base images:
screen shot 2016-02-17 at 4 51 41 pm

2. Host types:
screen shot 2016-02-17 at 4 52 03 pm

3. Security zones:
screen shot 2016-02-17 at 4 52 27 pm

4. Host Placements
screen shot 2016-02-17 at 4 52 39 pm

### CODE REVIEW

File: deploy-service/common/src/main/java/com/pinterest/clusterservice/cm/ClusterManager.java [hunk 1]
```diff
@@ -15,21 +15,20 @@
...
 public interface ClusterManager {
+    void createCluster(String clusterName, Object bean) throws Exception;
 
-    void createCluster(String clusterName, ClusterBean bean) throws Exception;
-
-    void updateCluster(String clusterName, ClusterBean bean) throws Exception;
+    void updateCluster(String clusterName, Object bean) throws Exception;
 
-    ClusterBean getCluster(String clusterName) throws Exception;
+    Object getCluster(String clusterName) throws Exception;
```
Comment (sbaogang) [inline deploy-service/common/src/main/java/com/pinterest/clusterservice/cm/ClusterManager.java]: We should not use Object, we can either use a common interface or bean (like you did before), or templatize this interface.

File: deploy-service/common/src/main/java/com/pinterest/clusterservice/common/ClusterConstants.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.clusterservice.common;
+
+
+public class ClusterConstants {
```
Comment (sbaogang) [inline deploy-service/common/src/main/java/com/pinterest/clusterservice/common/ClusterConstants.java]: I suggest to move these constants to AwsVmManager, they are AWS specific, also usually we like to put constants to where it belongs to.

Comment (yujunglo) [inline deploy-service/common/src/main/java/com/pinterest/clusterservice/common/ClusterConstants.java]: The PROVIDER_AWS_VM is a common constants. In the next version, there will be PROVIDER_AWS_CONTAINER, PROVIDER_GOOGLE_CONTAINER, etc.

**Comment by yujunglo** [comment]
Make Cloud Provider obvious.
screen shot 2016-02-20 at 8 29 52 pm

screen shot 2016-02-20 at 8 33 00 pm

**Comment by sbaogang** [comment]
r+

**Pull request merged**